system-tests/e2e-testsupport/src/main/java/com/hotels/styx/servers/MockOriginServer.java

```diff
@@ -0,0 +1,218 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hotels.styx.servers;
+
+import com.github.tomakehurst.wiremock.client.MappingBuilder;
+import com.github.tomakehurst.wiremock.client.RequestPatternBuilder;
+import com.github.tomakehurst.wiremock.client.ResponseDefinitionBuilder;
+import com.github.tomakehurst.wiremock.client.UrlMatchingStrategy;
+import com.github.tomakehurst.wiremock.client.WireMock;
+import com.github.tomakehurst.wiremock.core.WireMockApp;
+import com.github.tomakehurst.wiremock.core.WireMockConfiguration;
+import com.github.tomakehurst.wiremock.global.NotImplementedRequestDelayControl;
+import com.github.tomakehurst.wiremock.http.AdminRequestHandler;
+import com.github.tomakehurst.wiremock.http.BasicResponseRenderer;
+import com.github.tomakehurst.wiremock.http.ProxyResponseRenderer;
+import com.github.tomakehurst.wiremock.http.Request;
+import com.github.tomakehurst.wiremock.http.RequestHandler;
+import com.github.tomakehurst.wiremock.http.StubRequestHandler;
+import com.github.tomakehurst.wiremock.http.StubResponseRenderer;
+import com.google.common.collect.ImmutableList;
+import com.google.common.util.concurrent.ServiceManager;
+import com.hotels.styx.api.HttpHandler2;
+import com.hotels.styx.server.HttpConnectorConfig;
+import com.hotels.styx.server.HttpServer;
+import com.hotels.styx.server.HttpServers;
+import com.hotels.styx.server.HttpsConnectorConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.configureFor;
+import static com.github.tomakehurst.wiremock.client.WireMock.stubFor;
+import static com.google.common.base.Optional.absent;
+import static com.hotels.styx.servers.WiremockResponseConverter.toStyxResponse;
+import static java.lang.String.format;
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static java.util.Collections.emptyMap;
+import static rx.Observable.just;
+
+public final class MockOriginServer {
+    private static final String FILES_ROOT = "__files";
+    private static final Logger LOGGER = LoggerFactory.getLogger(MockOriginServer.class);
+    private static final int MAX_CONTENT_LENGTH = 256 * 1024;
+
+    private final String appId;
+    private final String originId;
+    private final int adminPort;
+
+    private int serverPort;
+    private final HttpServer adminServer;
+    private final HttpServer mockServer;
+
+    static {
+        System.setProperty("org.mortbay.log.class", "com.github.tomakehurst.wiremock.jetty.LoggerAdapter");
```

Comment:
Hmm... TBH I am not sure what this is supposed to do. I believe it can be removed. However I don't know if this would affect any logging originated by the Wiremock classes. 

We should follow up later.